src/praisonai-agents/praisonaiagents/memory/enhanced_memory.py

```
@@ -0,0 +1,647 @@
+        elif provider_lower == "redis":
+            return RedisStorage(config)
+        elif provider_lower == "dynamodb":
+            return DynamoDBStorage(config)
+        elif provider_lower == "s3":
+            return S3Storage(config)
+        elif provider_lower == "gcs":
+            return GCSStorage(config)
+        elif provider_lower == "azure":
+            return AzureStorage(config)
+        else:
+            raise ValueError(f"Unsupported storage provider: {provider}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)